scipy/_lib/_array_api.py

```diff
@@ -40,19 +41,29 @@
 # To enable array API and strict array-like input validation
 SCIPY_ARRAY_API: str | bool = os.environ.get("SCIPY_ARRAY_API", False)
 # To control the default device - for use in the test suite only
-SCIPY_DEVICE = os.environ.get("SCIPY_DEVICE", "cpu")
+SCIPY_DEVICE: str = os.environ.get("SCIPY_DEVICE", "cpu")
 
-_GLOBAL_CONFIG = {
-    "SCIPY_ARRAY_API": SCIPY_ARRAY_API,
-    "SCIPY_DEVICE": SCIPY_DEVICE,
-}
+# _GLOBAL_CONFIG = {
+#     "SCIPY_ARRAY_API": SCIPY_ARRAY_API,
+#     "SCIPY_DEVICE": SCIPY_DEVICE,
+# }
+
+# To enable changing the config programmatically
+@dataclass
+class GLOBAL_CONFIG:
+    SCIPY_ARRAY_API: str | bool = SCIPY_ARRAY_API
+    SCIPY_DEVICE: str = SCIPY_DEVICE
 
 
 if TYPE_CHECKING:
     Array = Any  # To be changed to a Protocol later (see array-api#589)
     ArrayLike = Array | npt.ArrayLike
 
 
+def set_scipy_array_api(val: str | bool) -> None:
+    GLOBAL_CONFIG.SCIPY_ARRAY_API = val
```

Comment:
Why expose a config setter without a matching getter? I would expect to either expose both a getter and a setter (and document this as the official way to interact with the config) or just expose the dataclass directly as the official way to interact with the config state.

Reply:
What is the case for allowing `val` to be of type `str`? Any non-empty string (e.g. "0") would be interpreted as `True` in expressions such as `if not GLOBAL_CONFIG.SCIPY_ARRAY_API` used in the code.